public/components/userpage.js

```diff
@@ -0,0 +1,100 @@
+
+ajaxRequest("GET", window.location + "?ajax=true", {}, {}, renderComponent);
+
+function renderComponent(data) {
+
+  // sort user freelancers by avgScore
+  // also call the variable "freelancers" because porcoddue sennó poi non capisco
+  let freelancers = data.freelancer;
+  freelancers.sort(function(a, b) {
+    return b.avgScore - a.avgScore;
+  });
+
+  let username = data.username;
+  const loggedUser = document.getElementById('react-user-page-root').getAttribute('data-username');
+  if (loggedUser && loggedUser == username) {
+    username = "you";
+  }
+
+  ReactDOM.render(
+    <UserPage
+      user={username}
+      freelancers={freelancers}
+    />,
+    document.getElementById('react-user-page-root')
+  );
+}
+
+class UserPage extends React.Component {
+  render() {
+    let freelancersList = [];
+    for (var i = 0; i < this.props.freelancers.length; i++) {
+      let freelancer = this.props.freelancers[i];
+      let categoryName = 'Other';
+      let categoryId = '';
+      if (freelancer.category) {
+        categoryName = freelancer.category.categoryName;
+        categoryId = freelancer.category._id;
+      }
+      freelancersList.push(<FreelancerCard
```

Comment:
filter by freelancer status please (i.e. do not show or flag the freelancer with claim 'in progress')